kube-client/src/client/config_ext.rs

```diff
@@ -182,6 +188,31 @@ impl ConfigExt for Config {
         })
     }
 
+    fn extra_headers_layer(&self) -> Result<ExtraHeadersLayer> {
+        let mut headers = Vec::new();
+        if let Some(impersonate_user) = &self.auth_info.impersonate {
+            headers.push((
+                HeaderName::from_static("impersonate-user"),
+                HeaderValue::from_str(impersonate_user)
+                    .map_err(http::Error::from)
+                    .map_err(Error::HttpError)?,
```

Comment:
Might be better to add an explicit error for invalid header value with the name.

`Error::HttpError` is only used by these

https://github.com/kube-rs/kube-rs/blob/18b5316b3a644a22ce64806b6cd2ed75d5e80b03/kube-client/src/client/mod.rs#L340-L345

and I think it should be removed eventually.

Reply:
Agreed, but that also feels like a part of the larger error refactoring.

Reply:
Yeah, we can do it later.